Source/Scene/ClassificationPrimitive.js

```diff
@@ -0,0 +1,916 @@
+        '../Core/isArray',
+        '../Renderer/DrawCommand',
+        '../Renderer/Pass',
+        '../Renderer/RenderState',
+        '../Renderer/ShaderProgram',
+        '../Renderer/ShaderSource',
+        '../Shaders/ShadowVolumeFS',
+        '../Shaders/ShadowVolumeVS',
+        '../ThirdParty/when',
+        './BlendingState',
+        './DepthFunction',
+        './PerInstanceColorAppearance',
+        './Primitive',
+        './SceneMode',
+        './StencilFunction',
+        './StencilOperation'
+    ], function(
+        ColorGeometryInstanceAttribute,
+        defaultValue,
+        defined,
+        defineProperties,
+        destroyObject,
+        DeveloperError,
+        GeometryInstance,
+        isArray,
+        DrawCommand,
+        Pass,
+        RenderState,
+        ShaderProgram,
+        ShaderSource,
+        ShadowVolumeFS,
+        ShadowVolumeVS,
+        when,
+        BlendingState,
+        DepthFunction,
+        PerInstanceColorAppearance,
+        Primitive,
+        SceneMode,
+        StencilFunction,
+        StencilOperation) {
+    'use strict';
+
+    var ClassificationPrimitiveReadOnlyInstanceAttributes = ['color'];
+
+    /**
+     * A classification primitive represents a volume enclosing geometry in the {@link Scene} to be highlighted.  The geometry must be from a single {@link GeometryInstance}.
+     * Batching multiple geometries is not yet supported.
+     * <p>
+     * A primitive combines the geometry instance with an {@link Appearance} that describes the full shading, including
+     * {@link Material} and {@link RenderState}.  Roughly, the geometry instance defines the structure and placement,
+     * and the appearance defines the visual characteristics.  Decoupling geometry and appearance allows us to mix
+     * and match most of them and add a new geometry or appearance independently of each other. Only the {@link PerInstanceColorAppearance}
+     * is supported at this time.
+     * </p>
+     * <p>
+     * For correct rendering, this feature requires the EXT_frag_depth WebGL extension. For hardware that do not support this extension, there
+     * will be rendering artifacts for some viewing angles.
+     * </p>
+     * <p>
+     * Valid geometries are {@link BoxGeometry}, {@link CylinderGeometry}, {@link EllipsoidGeometry}, {@link PolylineVolumeGeometry}, and {@link SphereGeometry}.
+     * </p>
+     * <p>
+     * Geometries that follow the surface of the ellipsoid, such as {@link CircleGeometry}, {@link CorridorGeometry}, {@link EllipseGeometry}, {@link PolygonGeometry}, and {@link RectangleGeometry},
+     * are also valid if they are extruded volumes; otherwise, they will not be rendered.
+     * </p>
+     *
+     * @alias ClassificationPrimitive
+     * @constructor
+     *
+     * @param {Object} [options] Object with the following properties:
+     * @param {Array|GeometryInstance} [options.geometryInstances] The geometry instances to render.
```

Comment:
This should be of length 1, right?  Mention it.

Leave it plural so we could easily relax the requirement later.